src/widgets/dialogs/LastRunCrashDialog.cpp

```diff
@@ -1,93 +1,162 @@
+    label->setWordWrap(true);
+
+    layout->addSpacing(16);
+
+    auto buttons = layout.emplace<QDialogButtonBox>();
+
+    auto *okButton = buttons->addButton(u"Ok"_s, QDialogButtonBox::AcceptRole);
     QObject::connect(okButton, &QPushButton::clicked, [this] {
         this->accept();
     });
 
-    // Updates
-    //    auto updateUpdateLabel = [update]() mutable {
-    //        auto &updateManager = UpdateManager::instance();
-
-    //        switch (updateManager.getStatus()) {
-    //            case UpdateManager::None: {
-    //                update->setText("Not checking for updates.");
-    //            } break;
-    //            case UpdateManager::Searching: {
-    //                update->setText("Checking for updates...");
-    //            } break;
-    //            case UpdateManager::UpdateAvailable: {
-    //                update->setText("Update available.");
-    //            } break;
-    //            case UpdateManager::NoUpdateAvailable: {
-    //                update->setText("No update abailable.");
-    //            } break;
-    //            case UpdateManager::SearchFailed: {
-    //                update->setText("Error while searching for update.\nEither
-    //                the update service is "
-    //                                "temporarily down or there is an issue
-    //                                with your installation.");
-    //            } break;
-    //            case UpdateManager::Downloading: {
-    //                update->setText(
-    //                    "Downloading the update. Chatterino will close once
-    //                    the download is done.");
-    //            } break;
-    //            case UpdateManager::DownloadFailed: {
-    //                update->setText("Download failed.");
-    //            } break;
-    //            case UpdateManager::WriteFileFailed: {
-    //                update->setText("Writing the update file to the hard drive
-    //                failed.");
-    //            } break;
-    //        }
-    //    };
-
-    //    updateUpdateLabel();
-    //    this->signalHolder_.managedConnect(updateManager.statusUpdated,
-    //    [updateUpdateLabel](auto) mutable {
-    //        postToThread([updateUpdateLabel]() mutable { updateUpdateLabel();
-    //        });
-    //    });
+    if (!getArgs().safeMode)
+    {
+        auto *safeModeButton = buttons->addButton(u"Restart in safe mode"_s,
+                                                  QDialogButtonBox::NoRole);
+        safeModeButton->setToolTipDuration(0);
+        safeModeButton->setToolTip(
+            u"In safe mode, the settings button is always shown"_s
+#ifdef CHATTERINO_HAVE_PLUGINS
+            " and plugins are disabled"
+#endif
+            ".\nSame as starting with --safe-mode.");
+
+        QObject::connect(safeModeButton, &QPushButton::clicked, [this] {
+            auto args = getArgs().currentArguments();
+            args += u"--safe-mode"_s;
```

Comment:
if `args` is a list, use `.append` or whatever function here instead of `+=` for clarity